packages/vm/src/compatibility.rs

```diff
@@ -81,7 +81,8 @@ const MAX_FUNCTION_RESULTS: usize = 1;
 
 /// Checks if the data is valid wasm and compatibility with the CosmWasm API (imports and exports)
 pub fn check_wasm(wasm_code: &[u8], available_capabilities: &HashSet<String>) -> VmResult<()> {
-    let module = ParsedWasm::parse(wasm_code)?;
+    let mut module = ParsedWasm::parse(wasm_code)?;
+    module.validate_funcs()?;
```

Comment:
Nice! Now let's move this line behind all the other more trivial check, right before `Ok(())`